vendor/orchestra-event-observer/src/indexer/stacks/blocks_pool.rs

```diff
@@ -0,0 +1,902 @@
+            if block_appended {
+                if let Some(new_fork) = new_fork.take() {
+                    let fork_id = self.forks.len();
+                    self.forks.insert(fork_id, new_fork);
+                    fork_updated = self.forks.get_mut(&fork_id);
+                } else {
+                    fork_updated = Some(fork);
+                }
+                // A block can only be added to one segment
+                break;
+            }
+        }
+
+        let fork_updated = match fork_updated.take() {
+            Some(fork) => {
+                info!(
+                    "Block {} successfully appended to {}",
+                    block.block_identifier, fork
+                );
+                fork
+            }
+            None => {
+                info!(
+                    "Unable to process block {} - inboxed for later",
+                    block.block_identifier
+                );
+                self.orphans.insert(block.block_identifier.clone());
+                return Ok(None);
+            }
+        };
+
+        // Process former orphans
+        let orphans = self.orphans.clone();
+        let mut orphans_to_untrack = HashSet::new();
+
+        let mut at_least_one_orphan_appended = true;
+        // As long as we are successful appending blocks that were previously unprocessable,
+        // Keep looping on this backlog
+        let mut applied = HashSet::new();
+        while at_least_one_orphan_appended {
+            at_least_one_orphan_appended = false;
+            for orphan_block_identifier in orphans.iter() {
+                if applied.contains(orphan_block_identifier) {
+                    continue;
+                }
+                let block = match self.block_store.get(orphan_block_identifier) {
+                    Some(block) => block.clone(),
+                    None => continue,
+                };
+
+                let (orphan_appended, new_fork) = fork_updated.try_append_block(&block);
+                if orphan_appended {
+                    applied.insert(orphan_block_identifier);
+                    orphans_to_untrack.insert(orphan_block_identifier);
+                }
+                at_least_one_orphan_appended = at_least_one_orphan_appended || orphan_appended;
+            }
+        }
+
+        // Update orphans
+        for orphan in orphans_to_untrack.into_iter() {
+            info!("Dequeuing orphan {}", orphan);
+            self.orphans.remove(orphan);
+        }
+
+        // Select canonical fork
+        let mut canonical_fork_id = 0;
+        let mut highest_height = 0;
+        let mut highest_btc_spent = 0;
+        for (fork_id, fork) in self.forks.iter() {
+            info!("Active fork: {} - {}", fork_id, fork);
+            if fork.get_length() >= highest_height {
+                highest_height = fork.get_length();
+                if fork.amount_of_btc_spent > highest_btc_spent
```

Comment:
Where does `amount_of_btc_spent` get set?

Reply:
Good catch, I'll handle this in the next iteration - https://github.com/hirosystems/orchestra/issues/5